mitodl/micromasters
# Added flow typechecking for JS

#### What are the relevant tickets?

closes #522 (I guess?)

#### What's this PR do?

This adds the flow typechecker to our project. This comprises a few changes:

- adding the package for the typechecker itself (flow-bin)
- a babel plugin to automatically strip out flow type annotations
- two eslint plugins to handle imported types correctly
- adding a bunch of type annotations to our code

#### Where should the reviewer start?

#### How should this be manually tested?

Make sure nothing is broken! I think I only added annotations and didn't change any functionality, but it's a little hard to be sure.

You'll need to rebuild your docker containers for the typechecker to work, once you have you should be able to do:

```
docker-compose run watch npm run-script flow
```

to type-check the code.

#### Any background context you want to provide?

#### Screenshots (if appropriate)

#### What GIF best describes this PR or how it makes you feel?
